### PR TITLE
APPSRE-10615 more DTP metrics

### PR DESCRIPTION
**Pull request by fishi0x01**
- Fix cluster metric -> previously we also counted filtered clusters
- add token gauge -> `dtp_tokens_managed{ ocm_env_name, dt_tenant_id }`

### CODE REVIEW

File: reconcile/dynatrace_token_provider/integration.py [hunk 1]
```diff
@@ -69,7 +71,30 @@ def run(self, dry_run: bool) -> None:
...
+                    DTPTokensManagedGauge(
+                        integration=self.name,
+                        ocm_env=ocm_env_name,
+                        dt_tenant_id=dt_tenant_id,
+                    ),
+                    cnt,
+                )
+
     def reconcile(self, dry_run: bool, dependencies: Dependencies) -> None:
+        self._lock = Lock()
```
Comment (hemslo) [inline reconcile/dynatrace_token_provider/integration.py:95]: why use lock, which part is run in threads?

File: reconcile/dynatrace_token_provider/integration.py [hunk 2]
```diff
@@ -69,7 +71,30 @@ def run(self, dry_run: bool) -> None:
         dependencies.populate()
         self.reconcile(dry_run=dry_run, dependencies=dependencies)
 
+    def _token_cnt(self, dt_tenant_id: str, ocm_env_name: str) -> None:
+        with self._lock:
+            if ocm_env_name not in self._managed_tokens_cnt:
+                self._managed_tokens_cnt[ocm_env_name] = {}
+            if dt_tenant_id not in self._managed_tokens_cnt[ocm_env_name]:
+                self._managed_tokens_cnt[ocm_env_name][dt_tenant_id] = 0
+            self._managed_tokens_cnt[ocm_env_name][dt_tenant_id] += 1
```
Comment (hemslo) [inline reconcile/dynatrace_token_provider/integration.py:80]: by using `defaultdict` and `Counter`, all key checks can be saved

first init `_managed_tokens_cnt` as `defaultdict`

```python
    self._managed_tokens_cnt = defaultdict(Counter)
```

then start count directly

```python
    self._managed_tokens_cnt[ocm_env_name][dt_tenant_id] += 1
```

Comment (fishi0x01) [reply reconcile/dynatrace_token_provider/integration.py]: Good idea! Fixed

File: reconcile/dynatrace_token_provider/integration.py (hunk 1 above)
Comment (fishi0x01) [reply reconcile/dynatrace_token_provider/integration.py:92]: Currently single threaded, but I want to keep this thread-safe as it is quite possible that we might need concurrency.

Comment (hemslo) [reply reconcile/dynatrace_token_provider/integration.py:92]: ok, then please move instance vars init (`_lock` and `_managed_tokens_cnt`) to `__init__` so we can clearly see what vars are available.

**Review by hemslo — APPROVED** [review]
LGTM!

File: reconcile/dynatrace_token_provider/integration.py (hunk 1 above)
Comment (fishi0x01) [reply reconcile/dynatrace_token_provider/integration.py]: Done :+1:

**Pull request merged**